controllers/controller.go

```diff
@@ -54,6 +55,18 @@ type EtcdadmClusterReconciler struct {
 	Scheme                  *runtime.Scheme
 	etcdHealthCheckConfig   etcdHealthCheckConfig
 	MaxConcurrentReconciles int
+	GetEtcdClient           func(ctx context.Context, cluster *clusterv1.Cluster, endpoints string) (EtcdClient, error)
+	isPortOpen              func(ctx context.Context, endpoint string) bool
+}
+
+type EtcdClient interface {
+	Close() error
+	MemberList(ctx context.Context) (*clientv3.MemberListResponse, error)
+	MemberRemove(ctx context.Context, id uint64) (*clientv3.MemberRemoveResponse, error)
+}
```

Comment:
Need to outline in a code comment why this exists. Similarly, we need to clearly state the consumer is responsible for calling `Close()` etc. Make it clear how this is intended to be used and that, generally, we expect it to to come from the etcd client lib.